### PR TITLE
create and destroy unit test

### PR DESCRIPTION
**Pull request by Xander3359**
unit test box: https://github.com/tgstation/tgstation/pull/57408/files
\+ Create_and_destroy unit test from TG

### CODE REVIEW

**Comment by LudwigVonChesterfield** [comment]
prepare for nobody running the unit tests ever

File: code/modules/unit_tests/create_and_destroy.dm [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+			oldest_packet_creation = min(qdeld_at, oldest_packet_creation)
+
+		//If we've found a packet that got del'd later then we finished, then all our shit has been processed
+		//That said, if there are any pending hard deletes you may NOT sleep, we gotta handle that shit
+		if(oldest_packet_creation > start_time && !length(SSgarbage.queues[GC_QUEUE_HARDDELETE]))
+			garbage_queue_processed = TRUE
+			break
+
+		if(REALTIMEOFDAY > real_start_time + time_needed + 50 MINUTES) //If this gets us gitbanned I'm going to laugh so hard
```
Comment (LudwigVonChesterfield) [inline code/modules/unit_tests/create_and_destroy.dm:85]: We might want to cut this time down to something more reasonably to run locally, like 10 minutes
```suggestion
		if(REALTIMEOFDAY > real_start_time + time_needed + 10 MINUTES) //If this gets us gitbanned I'm going to laugh so hard
```

File: code/modules/unit_tests/create_and_destroy.dm [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+			TEST_FAIL("Something has gone horribly wrong, the garbage queue has been processing for well over 30 minutes. What the hell did you do")
```
Comment (LudwigVonChesterfield) [inline code/modules/unit_tests/create_and_destroy.dm:86]:
```suggestion
			TEST_FAIL("Something has gone horribly wrong, the garbage queue has been processing for well over 10 minutes. What the hell did you do")
```

Comment (LudwigVonChesterfield) [inline code/modules/unit_tests/create_and_destroy.dm:86]:
```suggestion
			TEST_FAIL("Something has gone horribly wrong, the garbage queue has been processing for well over 50 minutes. What the hell did you do")
```

just to make it consistent with the line above

File: roguetown.dme [hunk 3]
```diff
@@ -8,8 +8,10 @@
 // END_PREFERENCES
 // BEGIN_INCLUDE
 #include "_maps\_basemap.dm"
+#include "_maps\templates\unit_tests.dmm"
```
Comment (JTGSZ) [inline roguetown.dme:11]: Might want to untick the stack of 7 empty space z-levels

Comment (JTGSZ) [reply roguetown.dme]: nvm i just did it for you

**Pull request merged**